pkg/operator/ceph/object/status.go

```diff
@@ -84,10 +84,11 @@ func updateStatusBucket(client client.Client, name types.NamespacedName, phase c
 
 func buildStatusInfo(cephObjectStore *cephv1.CephObjectStore) map[string]string {
 	m := make(map[string]string)
-	m["endpoint"] = buildDNSEndpoint(BuildDomainName(cephObjectStore.Name, cephObjectStore.Namespace), cephObjectStore.Spec.Gateway.Port, false)
 
 	if cephObjectStore.Spec.Gateway.SecurePort != 0 {
-		m["secureEndpoint"] = buildDNSEndpoint(BuildDomainName(cephObjectStore.Name, cephObjectStore.Namespace), cephObjectStore.Spec.Gateway.SecurePort, true)
+		m["endpoint"] = buildDNSEndpoint(BuildDomainName(cephObjectStore.Name, cephObjectStore.Namespace), cephObjectStore.Spec.Gateway.SecurePort, true)
```

Comment:
Please write unit tests for this function. We also need to handle the case where both are specified and then we have `m["endpoint"]` as well as `m["secureEndpoint"]`

Reply:
Done.